crates/node/src/asset_manager/mod.rs

```diff
@@ -115,21 +114,16 @@ impl AssetManager {
     }
 
     async fn process_program(&self, program: &Program) -> Result<()> {
-        // TODO: Process and program files are now downloaded in different ways. Combine these.
-        let file_path = PathBuf::new()
-            .join(self.config.data_directory.clone())
-            .join("images")
-            .join(program.hash.to_string())
-            .join(program.image_file_name.clone());
-
-        let url = reqwest::Url::parse(&program.image_file_url)?;
-        self.download(&url, &file_path).await
+        self.download_image(
+            &program.image_file_url,
+            program.hash,
+            &program.image_file_name,
+            &program.image_file_checksum,
+        )
```

Comment:
This function is always used just for programs. Maybe change the function param to take `&Program` ?

Reply:
Yes and download image will always come from a program.